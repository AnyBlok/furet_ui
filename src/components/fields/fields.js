export const fields = {
  list: {},
  form: {},
  thumbnail: {},
  kanban: {},
};
export default {
    fields,
}
