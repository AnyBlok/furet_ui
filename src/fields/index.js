/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import React from 'react';
import plugin from '../plugin';
import './string';
import './selection';
import './date';
import './integer';
import './float';
import './url';
import './uuid';
import './sequence';
import './password';
import './color';
import './text';

/**
 * Unknown field, used only if the wanted field is not available
**/
plugin.set(['field'], {Unknown: (props) => {
    return <div>{props.value}</div>
}});

/**
 * return the component for viewType and fieldType
**/
export const getField = (viewType, fieldType, attribs, value) => {
    let field = plugin.get(['field', viewType, fieldType]);
    if (!field) {
        field = plugin.get(['field', 'Unknown']);
        console.log('getField', viewType, fieldType, attribs, value)
    }
    const id = 'view-type-' + viewType + '-field_type-' + fieldType + '-field-name-' + (attribs.name || '');
    return React.createElement(field, Object.assign({id}, attribs, {value}));
}

export default {
    getField,
}
