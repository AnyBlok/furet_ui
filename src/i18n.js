/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import VueI18n from 'vue-i18n';
Vue.use(VueI18n);


export const i18nConf = {
    locale: 'en',
    fallbackLocale: 'en',
    messages: {
        en: {
            menus: {
                close: 'Close',
                search: 'Search ...',
            },
            views: {
                unknown: {
                    title: 'The wanted view "{name}" is unknown',
                    message: "Report this message to the administrator",
                },
                common: {
                    create: 'Create',
                    save: 'Save',
                    edit: 'Edit',
                    cancel: 'Cancel',
                    delete: 'Delete',
                    close: 'Close',
                    actions: 'Actions',
                    more: 'More',
                },
                clients: {
                    login: {
                        button: 'Log in',
                    },
                },
            },
            fields: {
                common: {
                    required: 'this field is required',
                },
                json: {
                    invalid: 'Invalid JSON format',
                },
                file: {
                    upload: 'Upload',
                    download: 'Download',
                    delete: 'Delete',
                },
            },
        },
    },
};
export const i18n = new VueI18n(i18nConf);
