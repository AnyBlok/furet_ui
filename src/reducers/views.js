/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
export const defaultState = {}

export const views = (state = defaultState, view) => {
    const value = Object.assign({}, view);
    delete value.type;
    switch (view.type) {
        case 'UPDATE_VIEW':
            delete value.viewId;
            const values = {};
            values[view.viewId] = Object.assign({}, state[view.viewId], value);
            return Object.assign({}, state, values);
        case 'CLEAR_ACTION':
            return defaultState;
        default:
            return state
    }
}

export default views;
