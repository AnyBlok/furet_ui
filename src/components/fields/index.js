import { fields } from './fields';
import './common';
import './boolean';
import './datetime';
import './integer';
import './password';
import './timestamp';
import './selection';
import './string';

import './relationship';

import { defineComponent } from '../factory';

defineComponent('furet-ui-unknown-field', {
  template : `<div>Unknow Field <strong>{{config.type}}</strong></div>`,
  prototype: {
    props: ['config', 'data'],
  }
})

defineComponent('furet-ui-field', {
  template : `
    <component 
      v-bind:is="component" 
      v-bind:resource="resource" 
      v-bind:config="config" 
      v-bind:data="data"
    />
  `,
  prototype: {
    props: ['resource', 'config', 'data'],
    computed: {
      component () {
        return (fields[this.resource.type] || {})[this.config.type] || 'furet-ui-unknown-field'
      },
    },
  }
})
