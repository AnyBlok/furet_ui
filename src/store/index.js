/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import _ from 'underscore';
import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import * as getters from './getters';
import modules from './modules';
import {i18n} from '../i18n';
import {router} from '../routes';
import moment from 'moment';

Vue.use(Vuex)
const debug = process.env.NODE_ENV !== 'production'

export const storeDef = {
    actions,
    getters,
    modules,
    strict: debug,
};

export const store = new Vuex.Store(storeDef);

export const dispatchAll = (datas) => {
    _.each(datas, data => {
        if (data.type) {
            switch (data.type) {
                case 'UPDATE_LOCALES':
                    _.each(data.locales, locale => {
                        i18n.setLocaleMessage(locale.locale, locale.messages)
                    });
                    break
                case 'SET_LOCALE':
                    i18n.locale = data.locale;
                    moment.locale(data.locale);
                    break
                case 'UPDATE_ROUTE':
                    if (data.name) router.push({name: data.name, params: data.params});
                    else if (data.path) router.push({path: data.path});
                    break;
                default:
                    const type = data.type;
                    delete data.type;
                    const process = data.process || 'commit';
                    store[process](type, data);
            }
        }
    });
}

export default store;
