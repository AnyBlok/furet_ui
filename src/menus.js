/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import './picture';
import _ from 'underscore';
import {dispatchAll} from './store';
import {json_post} from './server-call';
        
export const selectCard = (type, router, store, card) => {
    const key = 'UPDATE_' + (type || '').toUpperCase() + '_MENU';
    switch (card.type) {
        case 'client':
            router.push({name: 'custom_view', params: {viewName: card.id}}); 
            break;
        case 'space':
            router.push({name: 'space', params: {spaceId: card.id}});
            store.commit(key, {
                value: {
                    label: card.label,
                    image: card.image,
                },
            });
            break;
    }
    store.commit('CLEAR_BREADSCRUMB');
    store.commit('CLEAR_CHANGE');
}

export const Menu = Vue.component('furet-ui-appbar-menu', {
    template: `
        <a class="nav-item" v-if="hasValue">
            <a class="button" v-on:click="isModalActive = true; searchText = ''">
                <furet-ui-picture 
                    v-bind:type="value.image.type" 
                    v-bind:value="value.image.value" 
                />
                <span> {{ value.label }} </span>
            </a>
            <b-modal :active.sync="isModalActive">
                <div class="modal-background"></div>
                <div class="modal-card">
                    <header class="modal-card-head">
                        <b-field  position="is-centered">
                            <b-input
                                type="search"
                                icon-pack="fa"
                                icon="search"
                                v-bind:placeholder="$t('menus.search')"
                                v-model="searchText">
                            </b-input>
                        </b-field>
                    </header>
                    <div 
                        class="modal-card-body"
                        v-bind:style="{color: '#363636', padding: '5px'}"
                    >
                        <div v-for="group in groups">
                            <legend>
                                <furet-ui-picture
                                    v-bind:type="group.image.type"
                                    v-bind:value="group.image.value"
                                />
                                {{ group.labelÂ }}
                            </legend>
                            <div class="columns is-multiline is-mobile">
                                <div class="column is-12-mobile is-half-tablet is-half-desktop"
                                    v-for="card in group.values">
                                        <article 
                                            class="box media" 
                                            v-on:click.stop="selectCard(card)"
                                            v-bind:style="{padding: '5px'}"
                                        >
                                            <div class="media-left">
                                                <figure class="image is-32x32">
                                                    <furet-ui-picture
                                                        v-bind:type="card.image.type"
                                                        v-bind:value="card.image.value"
                                                    />
                                                </figure>
                                            </div>
                                            <div class="media-content">
                                                <div >
                                                    <strong>{{card.label}}</strong>
                                                    <br />
                                                    <small v-bind:style="{whiteSpace: 'pre-wrap'}">{{card.description}}</small>
                                                </div>
                                            </div>
                                        </article>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <footer class="modal-card-foot">
                        <a 
                            v-on:click="isModalActive = false"
                            v-bind:style="{color: '#363636'}"
                        >{{$t('menus.close')}}</a>
                    </footer>
                </div>
            </b-modal>
        </a>`,
    props: ['type', 'unittest_active', 'unittest_search'],
    data () {
        return {
            isModalActive: this.unittest_active || false, 
            searchText: this.unittest_search || '',
        };
    },
    computed: {
        value () {
            return this.$store.state[this.type + 'menu'].value;
        },
        hasValue () {
            const value = this.$store.state[this.type + 'menu'].value,
                  hasImage = _.keys(value.image || {}).length > 0,
                  hasLabel = value.label.length > 0;
            return hasImage && hasLabel;
        },
        groups () {
            const groups = [],
                  re = new RegExp(this.searchText, 'ig');
            _.each(this.$store.state[this.type + 'menu'].values, g => {
                const cards = _.filter(g.values, c => (re.test(c.label) || re.test(c.description)))
                if (cards.length) groups.push(Object.assign({}, g, {values: cards}));
            });
            return groups;
        },
    },
    methods: {
        selectCard (card) {
            selectCard(this.type, this.$router, this.$store, card);
            this.isModalActive = false;
        }
    }
});

export const LeftMenu = Vue.component('furet-ui-appbar-left-menu', {
    template: '<furet-ui-appbar-menu type="left"/>',
});

export const RightMenu = Vue.component('furet-ui-appbar-right-menu', {
    template: '<furet-ui-appbar-menu type="right"/>',
});
