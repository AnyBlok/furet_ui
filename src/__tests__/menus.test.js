/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import Vuex from 'vuex';
import Buefy from 'buefy';
Vue.use(Buefy, {defaultIconPack: 'fa',});
import '../views';
import '../fields';
import {store} from '../store';
import {router} from '../routes';
import {i18n} from '../i18n';
import {Menu, LeftMenu, RightMenu, selectCard} from '../menus';

jest.mock('../server-call')

describe('LeftMenu component', () => {
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render Left Menu with default value from store', () => {
        const renderer = require('vue-server-renderer').createRenderer();
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(LeftMenu),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render Left Menu with value from store', () => {
        const renderer = require('vue-server-renderer').createRenderer();
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(LeftMenu),
        }).$mount();
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
        store.commit({
            'type': 'UPDATE_LEFT_MENU',
            'value': {
                'label': 'Login',
                'image': {'type': 'font-icon', 'value': 'fa-user'},
            },
            'values': [
                {
                    'label': 'Login',
                    'image': {'type': 'font-icon', 'value': 'fa-user'},
                    'id': 'login',
                    'values': [
                        {
                            'label': 'Login',
                            'description': 'Log in to use the application',
                            'image': {'type': 'font-icon', 'value': 'fa-user'},
                            'type': 'client',
                            'id': 'Login',
                        },
                        {
                            'label': 'test',
                            'description': '',
                            'image': {'type': 'font-icon', 'value': 'fa-user'},
                            'type': 'client',
                            'id': 'Login',
                        },
                    ],
                },
            ],
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});

describe('RightMenu component', () => {
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render Right Menu with default value from store', () => {
        const renderer = require('vue-server-renderer').createRenderer();
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(RightMenu),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render Right Menu with value from store', () => {
        const renderer = require('vue-server-renderer').createRenderer();
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(RightMenu),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
        store.commit({
            'type': 'UPDATE_RIGHT_MENU',
            'value': {
                'label': 'Login',
                'image': {'type': 'font-icon', 'value': 'fa-user'},
            },
            'values': [
                {
                    'label': 'Login',
                    'image': {'type': 'font-icon', 'value': 'fa-user'},
                    'id': 'login',
                    'values': [
                        {
                            'label': 'Login',
                            'description': 'Log in to use the application',
                            'image': {'type': 'font-icon', 'value': 'fa-user'},
                            'type': 'client',
                            'id': 'Login',
                        },
                        {
                            'label': '',
                            'description': '',
                            'image': {'type': 'font-icon', 'value': 'fa-user'},
                            'type': 'client',
                            'id': 'Login',
                        },
                    ],
                },
            ],
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
})
describe('Menu component', () => {
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render Menu with value from store open', () => {
        const renderer = require('vue-server-renderer').createRenderer();
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Menu, {props: {type: 'left', unittest_active: true}}),
        }).$mount();
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
        store.commit({
            'type': 'UPDATE_LEFT_MENU',
            'value': {
                'label': 'Login',
                'image': {'type': 'font-icon', 'value': 'fa-user'},
            },
            'values': [
                {
                    'label': 'Login',
                    'image': {'type': 'font-icon', 'value': 'fa-user'},
                    'id': 'login',
                    'values': [
                        {
                            'label': 'Login',
                            'description': 'Log in to use the application',
                            'image': {'type': 'font-icon', 'value': 'fa-user'},
                            'type': 'client',
                            'id': 'Login',
                        },
                        {
                            'label': 'test',
                            'description': '',
                            'image': {'type': 'font-icon', 'value': 'fa-user'},
                            'type': 'client',
                            'id': 'Login',
                        },
                    ],
                },
            ],
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render Menu with value from store open and filtering', () => {
        const renderer = require('vue-server-renderer').createRenderer();
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Menu, {props: {type: 'left', unittest_active: true, unittest_search: 'test'}}),
        }).$mount();
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
        store.commit({
            'type': 'UPDATE_LEFT_MENU',
            'value': {
                'label': 'Login',
                'image': {'type': 'font-icon', 'value': 'fa-user'},
            },
            'values': [
                {
                    'label': 'Login',
                    'image': {'type': 'font-icon', 'value': 'fa-user'},
                    'id': 'login',
                    'values': [
                        {
                            'label': 'Login',
                            'description': 'Log in to use the application',
                            'image': {'type': 'font-icon', 'value': 'fa-user'},
                            'type': 'client',
                            'id': 'Login',
                        },
                        {
                            'label': 'test',
                            'description': '',
                            'image': {'type': 'font-icon', 'value': 'fa-user'},
                            'type': 'client',
                            'id': 'Login',
                        },
                    ],
                },
            ],
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render Menu with value from store open and select custom view', () => {
        const renderer = require('vue-server-renderer').createRenderer();
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Menu, {props: {type: 'left', unittest_active: true}}),
        }).$mount();
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
        store.commit({
            'type': 'UPDATE_LEFT_MENU',
            'value': {
                'label': 'Login',
                'image': {'type': 'font-icon', 'value': 'fa-user'},
            },
            'values': [
                {
                    'label': 'Login',
                    'image': {'type': 'font-icon', 'value': 'fa-user'},
                    'id': 'login',
                    'values': [
                        {
                            'label': 'Login',
                            'description': 'Log in to use the application',
                            'image': {'type': 'font-icon', 'value': 'fa-user'},
                            'type': 'client',
                            'id': 'Login',
                        },
                        {
                            'label': 'test',
                            'description': '',
                            'image': {'type': 'font-icon', 'value': 'fa-user'},
                            'type': 'client',
                            'id': 'Login',
                        },
                    ],
                },
            ],
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
        router.push({'path': '/'})
        expect(vm.$route.path).toBe('/');
        selectCard('left', router, store, {type: 'client', id: 'custom_view_id'});
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
        expect(vm.$route.name).toBe('custom_view');
        expect(vm.$route.params.viewName).toBe('custom_view_id');
    });
    it('Render Menu with value from store open and select space', () => {
        const renderer = require('vue-server-renderer').createRenderer();
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Menu, {props: {type: 'left', unittest_active: true}}),
        }).$mount();
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
        store.commit({
            'type': 'UPDATE_LEFT_MENU',
            'value': {
                'label': 'Login',
                'image': {'type': 'font-icon', 'value': 'fa-user'},
            },
            'values': [
                {
                    'label': 'Login',
                    'image': {'type': 'font-icon', 'value': 'fa-user'},
                    'id': 'login',
                    'values': [
                        {
                            'label': 'Login',
                            'description': 'Log in to use the application',
                            'image': {'type': 'font-icon', 'value': 'fa-user'},
                            'type': 'client',
                            'id': 'Login',
                        },
                        {
                            'label': 'test',
                            'description': '',
                            'image': {'type': 'font-icon', 'value': 'fa-user'},
                            'type': 'client',
                            'id': 'Login',
                        },
                    ],
                },
            ],
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
        router.push({'path': '/'})
        expect(vm.$route.path).toBe('/');
        store.commit('ADD_IN_BREADSCRUMB', {
            path: '/test',
            label: 'test',
            changes: {},
        });
        store.commit('REPLACE_CHANGE', {changes: {test: 1}})
        expect(vm.$store.state.global.breadscrumbs.length).toBe(1);
        expect(vm.$store.state.data.changes.test).toBe(1);
        selectCard('left', router, store, {type: 'space', id: 'spaceId', label: 'New Space', image: {type: 'font-icon', value: 'fa-trash'}});
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
        expect(vm.$route.name).toBe('space');
        expect(vm.$route.params.spaceId).toBe('spaceId');
        expect(vm.$store.state.global.breadscrumbs.length).toBe(0);
        expect(vm.$store.state.data.changes.test).toBe(undefined);
    });
})
