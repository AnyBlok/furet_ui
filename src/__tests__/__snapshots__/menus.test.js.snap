// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftMenu component Render Left Menu with default value from store 1`] = `<!---->`;

exports[`LeftMenu component Render Left Menu with value from store 1`] = `<!---->`;

exports[`LeftMenu component Render Left Menu with value from store 2`] = `
<a data-server-rendered="true" class="nav-item"><a class="button"><span class="icon"><i class="fa fa-user"></i></span> <span> Login </span></a>
<!---->
</a>
`;

exports[`Menu component Render Menu with value from store open 1`] = `<!---->`;

exports[`Menu component Render Menu with value from store open 2`] = `
<a data-server-rendered="true" class="nav-item"><a class="button"><span class="icon"><i class="fa fa-user"></i></span> <span> Login </span></a>
<div class="modal is-active">
    <div class="modal-background"></div>
    <div class="animation-content modal-content" style="max-width:960px;">
        <div class="modal-background"></div>
        <div class="modal-card">
            <header class="modal-card-head">
                <div class="field has-addons-centered">
                    <!---->
                    <div class="control has-icons-left is-clearfix"><input type="search" placeholder="Search ..." autocomplete="on" value="" class="input"> <span class="icon is-left"><i class="fa fa-search"></i></span>
                        <!---->
                        <!---->
                    </div>
                    <!---->
                </div>
            </header>
            <div class="modal-card-body" style="color:#363636;padding:5px;">
                <div>
                    <legend><span class="icon"><i class="fa fa-user"></i></span> Login
                    </legend>
                    <div class="columns is-multiline is-mobile">
                        <div class="column is-12-mobile is-half-tablet is-half-desktop">
                            <article class="box media" style="padding:5px;">
                                <div class="media-left">
                                    <figure class="image is-32x32"><span class="icon"><i class="fa fa-user"></i></span></figure>
                                </div>
                                <div class="media-content">
                                    <div><strong>Login</strong> <br> <small style="white-space:pre-wrap;">Log in to use the application</small></div>
                                </div>
                            </article>
                        </div>
                        <div class="column is-12-mobile is-half-tablet is-half-desktop">
                            <article class="box media" style="padding:5px;">
                                <div class="media-left">
                                    <figure class="image is-32x32"><span class="icon"><i class="fa fa-user"></i></span></figure>
                                </div>
                                <div class="media-content">
                                    <div><strong>test</strong> <br> <small style="white-space:pre-wrap;"></small></div>
                                </div>
                            </article>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        <footer class="modal-card-foot"><a style="color:#363636;">Close</a></footer>
    </div> <button class="modal-close is-large"></button></div>
</a>
`;

exports[`Menu component Render Menu with value from store open and filtering 1`] = `<!---->`;

exports[`Menu component Render Menu with value from store open and filtering 2`] = `
<a data-server-rendered="true" class="nav-item"><a class="button"><span class="icon"><i class="fa fa-user"></i></span> <span> Login </span></a>
<div class="modal is-active">
    <div class="modal-background"></div>
    <div class="animation-content modal-content" style="max-width:960px;">
        <div class="modal-background"></div>
        <div class="modal-card">
            <header class="modal-card-head">
                <div class="field has-addons-centered">
                    <!---->
                    <div class="control has-icons-left is-clearfix"><input type="search" placeholder="Search ..." autocomplete="on" value="test" class="input"> <span class="icon is-left"><i class="fa fa-search"></i></span>
                        <!---->
                        <!---->
                    </div>
                    <!---->
                </div>
            </header>
            <div class="modal-card-body" style="color:#363636;padding:5px;">
                <div>
                    <legend><span class="icon"><i class="fa fa-user"></i></span> Login
                    </legend>
                    <div class="columns is-multiline is-mobile">
                        <div class="column is-12-mobile is-half-tablet is-half-desktop">
                            <article class="box media" style="padding:5px;">
                                <div class="media-left">
                                    <figure class="image is-32x32"><span class="icon"><i class="fa fa-user"></i></span></figure>
                                </div>
                                <div class="media-content">
                                    <div><strong>test</strong> <br> <small style="white-space:pre-wrap;"></small></div>
                                </div>
                            </article>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        <footer class="modal-card-foot"><a style="color:#363636;">Close</a></footer>
    </div> <button class="modal-close is-large"></button></div>
</a>
`;

exports[`Menu component Render Menu with value from store open and select custom view 1`] = `<!---->`;

exports[`Menu component Render Menu with value from store open and select custom view 2`] = `
<a data-server-rendered="true" class="nav-item"><a class="button"><span class="icon"><i class="fa fa-user"></i></span> <span> Login </span></a>
<div class="modal is-active">
    <div class="modal-background"></div>
    <div class="animation-content modal-content" style="max-width:960px;">
        <div class="modal-background"></div>
        <div class="modal-card">
            <header class="modal-card-head">
                <div class="field has-addons-centered">
                    <!---->
                    <div class="control has-icons-left is-clearfix"><input type="search" placeholder="Search ..." autocomplete="on" value="" class="input"> <span class="icon is-left"><i class="fa fa-search"></i></span>
                        <!---->
                        <!---->
                    </div>
                    <!---->
                </div>
            </header>
            <div class="modal-card-body" style="color:#363636;padding:5px;">
                <div>
                    <legend><span class="icon"><i class="fa fa-user"></i></span> Login
                    </legend>
                    <div class="columns is-multiline is-mobile">
                        <div class="column is-12-mobile is-half-tablet is-half-desktop">
                            <article class="box media" style="padding:5px;">
                                <div class="media-left">
                                    <figure class="image is-32x32"><span class="icon"><i class="fa fa-user"></i></span></figure>
                                </div>
                                <div class="media-content">
                                    <div><strong>Login</strong> <br> <small style="white-space:pre-wrap;">Log in to use the application</small></div>
                                </div>
                            </article>
                        </div>
                        <div class="column is-12-mobile is-half-tablet is-half-desktop">
                            <article class="box media" style="padding:5px;">
                                <div class="media-left">
                                    <figure class="image is-32x32"><span class="icon"><i class="fa fa-user"></i></span></figure>
                                </div>
                                <div class="media-content">
                                    <div><strong>test</strong> <br> <small style="white-space:pre-wrap;"></small></div>
                                </div>
                            </article>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        <footer class="modal-card-foot"><a style="color:#363636;">Close</a></footer>
    </div> <button class="modal-close is-large"></button></div>
</a>
`;

exports[`Menu component Render Menu with value from store open and select custom view 3`] = `
<a data-server-rendered="true" class="nav-item"><a class="button"><span class="icon"><i class="fa fa-user"></i></span> <span> Login </span></a>
<div class="modal is-active">
    <div class="modal-background"></div>
    <div class="animation-content modal-content" style="max-width:960px;">
        <div class="modal-background"></div>
        <div class="modal-card">
            <header class="modal-card-head">
                <div class="field has-addons-centered">
                    <!---->
                    <div class="control has-icons-left is-clearfix"><input type="search" placeholder="Search ..." autocomplete="on" value="" class="input"> <span class="icon is-left"><i class="fa fa-search"></i></span>
                        <!---->
                        <!---->
                    </div>
                    <!---->
                </div>
            </header>
            <div class="modal-card-body" style="color:#363636;padding:5px;">
                <div>
                    <legend><span class="icon"><i class="fa fa-user"></i></span> Login
                    </legend>
                    <div class="columns is-multiline is-mobile">
                        <div class="column is-12-mobile is-half-tablet is-half-desktop">
                            <article class="box media" style="padding:5px;">
                                <div class="media-left">
                                    <figure class="image is-32x32"><span class="icon"><i class="fa fa-user"></i></span></figure>
                                </div>
                                <div class="media-content">
                                    <div><strong>Login</strong> <br> <small style="white-space:pre-wrap;">Log in to use the application</small></div>
                                </div>
                            </article>
                        </div>
                        <div class="column is-12-mobile is-half-tablet is-half-desktop">
                            <article class="box media" style="padding:5px;">
                                <div class="media-left">
                                    <figure class="image is-32x32"><span class="icon"><i class="fa fa-user"></i></span></figure>
                                </div>
                                <div class="media-content">
                                    <div><strong>test</strong> <br> <small style="white-space:pre-wrap;"></small></div>
                                </div>
                            </article>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        <footer class="modal-card-foot"><a style="color:#363636;">Close</a></footer>
    </div> <button class="modal-close is-large"></button></div>
</a>
`;

exports[`Menu component Render Menu with value from store open and select space 1`] = `<!---->`;

exports[`Menu component Render Menu with value from store open and select space 2`] = `
<a data-server-rendered="true" class="nav-item"><a class="button"><span class="icon"><i class="fa fa-user"></i></span> <span> Login </span></a>
<div class="modal is-active">
    <div class="modal-background"></div>
    <div class="animation-content modal-content" style="max-width:960px;">
        <div class="modal-background"></div>
        <div class="modal-card">
            <header class="modal-card-head">
                <div class="field has-addons-centered">
                    <!---->
                    <div class="control has-icons-left is-clearfix"><input type="search" placeholder="Search ..." autocomplete="on" value="" class="input"> <span class="icon is-left"><i class="fa fa-search"></i></span>
                        <!---->
                        <!---->
                    </div>
                    <!---->
                </div>
            </header>
            <div class="modal-card-body" style="color:#363636;padding:5px;">
                <div>
                    <legend><span class="icon"><i class="fa fa-user"></i></span> Login
                    </legend>
                    <div class="columns is-multiline is-mobile">
                        <div class="column is-12-mobile is-half-tablet is-half-desktop">
                            <article class="box media" style="padding:5px;">
                                <div class="media-left">
                                    <figure class="image is-32x32"><span class="icon"><i class="fa fa-user"></i></span></figure>
                                </div>
                                <div class="media-content">
                                    <div><strong>Login</strong> <br> <small style="white-space:pre-wrap;">Log in to use the application</small></div>
                                </div>
                            </article>
                        </div>
                        <div class="column is-12-mobile is-half-tablet is-half-desktop">
                            <article class="box media" style="padding:5px;">
                                <div class="media-left">
                                    <figure class="image is-32x32"><span class="icon"><i class="fa fa-user"></i></span></figure>
                                </div>
                                <div class="media-content">
                                    <div><strong>test</strong> <br> <small style="white-space:pre-wrap;"></small></div>
                                </div>
                            </article>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        <footer class="modal-card-foot"><a style="color:#363636;">Close</a></footer>
    </div> <button class="modal-close is-large"></button></div>
</a>
`;

exports[`Menu component Render Menu with value from store open and select space 3`] = `
<a data-server-rendered="true" class="nav-item"><a class="button"><span class="icon"><i class="fa fa-trash"></i></span> <span> New Space </span></a>
<div class="modal is-active">
    <div class="modal-background"></div>
    <div class="animation-content modal-content" style="max-width:960px;">
        <div class="modal-background"></div>
        <div class="modal-card">
            <header class="modal-card-head">
                <div class="field has-addons-centered">
                    <!---->
                    <div class="control has-icons-left is-clearfix"><input type="search" placeholder="Search ..." autocomplete="on" value="" class="input"> <span class="icon is-left"><i class="fa fa-search"></i></span>
                        <!---->
                        <!---->
                    </div>
                    <!---->
                </div>
            </header>
            <div class="modal-card-body" style="color:#363636;padding:5px;">
                <div>
                    <legend><span class="icon"><i class="fa fa-user"></i></span> Login
                    </legend>
                    <div class="columns is-multiline is-mobile">
                        <div class="column is-12-mobile is-half-tablet is-half-desktop">
                            <article class="box media" style="padding:5px;">
                                <div class="media-left">
                                    <figure class="image is-32x32"><span class="icon"><i class="fa fa-user"></i></span></figure>
                                </div>
                                <div class="media-content">
                                    <div><strong>Login</strong> <br> <small style="white-space:pre-wrap;">Log in to use the application</small></div>
                                </div>
                            </article>
                        </div>
                        <div class="column is-12-mobile is-half-tablet is-half-desktop">
                            <article class="box media" style="padding:5px;">
                                <div class="media-left">
                                    <figure class="image is-32x32"><span class="icon"><i class="fa fa-user"></i></span></figure>
                                </div>
                                <div class="media-content">
                                    <div><strong>test</strong> <br> <small style="white-space:pre-wrap;"></small></div>
                                </div>
                            </article>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        <footer class="modal-card-foot"><a style="color:#363636;">Close</a></footer>
    </div> <button class="modal-close is-large"></button></div>
</a>
`;

exports[`RightMenu component Render Right Menu with default value from store 1`] = `<!---->`;

exports[`RightMenu component Render Right Menu with value from store 1`] = `<!---->`;

exports[`RightMenu component Render Right Menu with value from store 2`] = `
<a data-server-rendered="true" class="nav-item"><a class="button"><span class="icon"><i class="fa fa-user"></i></span> <span> Login </span></a>
<!---->
</a>
`;
